apps/cowswap-frontend/src/modules/ordersTable/containers/OrdersTableWidget/hooks/useOrdersTableList.ts

```diff
@@ -53,33 +59,52 @@ export function useOrdersTableList(
         acc.all.push(item)
 
         const isPending = PENDING_STATES.includes(order.status)
+        const isSigning = order.status === OrderStatus.PRESIGNATURE_PENDING
 
         // Check if order is unfillable (insufficient balance or allowance)
         const params = getOrderParams(chainId, balancesAndAllowances, order)
         const isUnfillable = params.hasEnoughBalance === false || params.hasEnoughAllowance === false
 
-        // Only add to unfillable if the order is both pending and unfillable
-        if (isPending && isUnfillable) {
+        // Update the unfillable flag whenever the state changes, not just when becoming unfillable
+        if (isPending && order.isUnfillable !== isUnfillable) {
+          setIsOrderUnfillable({ chainId, id: order.id, isUnfillable })
+        }
+
+        // Add to signing if in presignature pending state
+        if (isSigning) {
+          acc.signing.push(item)
+        }
+
+        // Add to unfillable only if pending, unfillable, and not in signing state
+        if (isPending && isUnfillable && !isSigning) {
           acc.unfillable.push(item)
         }
 
         // Add to pending or history based on status
-        if (isPending) {
+        if (isPending && !isSigning) {
           acc.pending.push(item)
-        } else {
+        } else if (!isPending) {
           acc.history.push(item)
         }
 
         return acc
       },
-      { pending: [], history: [], unfillable: [], all: [] } as OrdersTableList,
+      {
+        pending: [] as OrderTableItem[],
```

Comment:
It's better to make `allSortedOrders.reduce<OrdersTableList>` instead of casting `as OrderTableItem[]` here